diff --git a/build-aux/ltmain.sh b/build-aux/ltmain.sh
index 21e5e078..ddd6e36f 100755
--- a/build-aux/ltmain.sh
+++ b/build-aux/ltmain.sh
@@ -7369,11 +7369,14 @@ func_mode_link ()
       # -fsanitize=*         Clang/GCC memory and address sanitizer
       # -fuse-ld=*           Linker select flags for GCC
       # -static-*            direct GCC to link specific libraries statically
+      # -shared-*            direct GCC to link shared version of specific libraries
       # -fcilkplus           Cilk Plus language extension features for C/C++
       -64|-mips[0-9]|-r[0-9][0-9]*|-xarch=*|-xtarget=*|+DA*|+DD*|-q*|-m*| \
       -t[45]*|-txscale*|-p|-pg|--coverage|-fprofile-*|-F*|@*|-tp=*|--sysroot=*| \
       -O*|-g*|-flto*|-fwhopr*|-fuse-linker-plugin|-fstack-protector*|-stdlib=*| \
-      -specs=*|-fsanitize=*|-fuse-ld=*|-static-*|-fcilkplus)
+      -specs=*|-fsanitize=*|-fuse-ld=*|-static-*|-shared-*|-fcilkplus| \
+      -ftree-parallelize-loops=*|-fgnu-tm|-ffast-math| \
+      -funsafe-math-optimizations|-fvtable-verify*)
         func_quote_for_eval "$arg"
 	arg=$func_quote_for_eval_result
         func_append compile_command " $arg"
